File: languages/php/Bloock/Entity/Identity/CredentialReceipt.php
<?php

namespace Bloock\Entity\Identity;

/**
 * Represents a receipt for a credential, including the credential itself, its ID, and type.
 */
class CredentialReceipt
{
    private Credential $credential;
    private string $credentialId;
    private string $credentialType;

    /**
     * Constructs an CredentialReceipt object with the specified parameters.
     * @param Credential $credential
     * @param string $credentialId
     * @param string $credentialType
     */
    public function __construct(Credential $credential, string $credentialId, string $credentialType)
    {
        $this->credential = $credential;
        $this->credentialId = $credentialId;
        $this->credentialType = $credentialType;
    }

    public static function fromProto(\Bloock\CredentialReceipt $res): CredentialReceipt
    {
        return new CredentialReceipt(Credential::fromProto($res->getCredential()), $res->getCredentialId(), $res->getCredentialType());
    }

    /**
     * Gets the credential object.
     * @return Credential
     */
    public function getCredential(): Credential
    {
        return $this->credential;
    }

    /**
     * Gets the ID associated with the credential.
     * @return string
     */
    public function getCredentialId(): string
    {
        return $this->credentialId;
    }

    /**
     * Gets the type of the credential.
     * @return string
     */
    public function getCredentialType(): string
    {
        return $this->credentialType;
    }

    public function toProto(): \Bloock\CredentialReceipt
    {
        $p = new \Bloock\CredentialReceipt();
        $p->setCredential($this->credential->toProto());
        $p->setCredentialId($this->credentialId);
        $p->setCredentialType($this->credentialType);
        return $p;
    }
}

File: languages/python/bloock/entity/identity/credential_receipt.py
from __future__ import annotations

import bloock._bridge.proto.identity_entities_pb2 as proto
from bloock.entity.identity.credential import Credential


class CredentialReceipt:
    """
    Represents a receipt for a credential, including the credential itself, its ID, and type.
    """
    def __init__(self, credential: Credential, credential_id: str, credential_type: str) -> None:
        """
        Constructs an CredentialReceipt object with the specified parameters.
        :type credential_type: object
        :type credential_id: object
        :type credential: object
        :rtype: object
        """
        self.credential = credential
        self.credential_id = credential_id
        self.credential_type = credential_type

    @staticmethod
    def from_proto(c: proto.CredentialReceipt) -> CredentialReceipt:
        return CredentialReceipt(
            credential=Credential.from_proto(c.credential),
            credential_id=c.credential_id,
            credential_type=c.credential_type,
        )

    def to_proto(self) -> proto.CredentialReceipt:
        return proto.CredentialReceipt(
            credential=self.credential.to_proto(),
            credential_id=self.credential_id,
            credential_type=self.credential_type,
        )
